com.unity.shadergraph/Editor/Data/Graphs/GraphData.cs

```diff
@@ -1937,21 +2060,26 @@ void UpgradeFromBlockMap(Dictionary<BlockFieldDescriptor, int> blockMap)
                     m_VertexContext.position = oldPosition;
                     m_FragmentContext.position = new Vector2(oldPosition.x, oldPosition.y + 200);
 
-                    // Try to upgrade all valid targets from master node
-                    // On ShaderGraph side we dont know what Targets exist so make no assumptions
-                    if(masterNode != null)
+                    // Try to upgrade all potential targets from master node
+                    if (masterNode != null)
                     {
-                        foreach(var target in m_ValidTargets)
+                        foreach (var potentialTarget in m_AllPotentialTargets)
                         {
-                            if(!(target is ILegacyTarget legacyTarget))
+                            if (potentialTarget.IsUnknown())
```

Comment:
are these equivalent? It feels like the ! legacy target catches more cases

Reply:
The corresponding !legacy target is below -- github's diff is not lining up the comparison correctly.

Basically we should never get an unknown target in m_AllPotentialTargets at this point ,as we're coming off of a deserialization, m_AllPotentialTargets should just be the reflection-initialized set of known Targets.   I even assert that is the case below...   we could just copy that assert up here as well and drop this check.